# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fs/child/FileSystemManagerChild.h
# Commit: b52f1a06e743
# Full Hash: b52f1a06e743ffb7695d349cc7edf11d0b5ce80e
# Author: Randell Jesup <rjesup@wgate.com>
# Date: 2022-09-10 21:45:26
# Regressor Bug: 1786465
# File Overlap Count: 1
# Description:
#   Bug 1786465 - Simplify FileSystemManagerChild by using the VirtualSendImpl attribute in PFileSystemManager definition; r=dom-storage-reviewers,jesup
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D155352
# ==============================================================================

diff -r c4555c4937a9 -r b52f1a06e743 dom/fs/child/FileSystemManagerChild.h
--- a/dom/fs/child/FileSystemManagerChild.h	Sat Sep 10 04:35:48 2022 +0000
+++ b/dom/fs/child/FileSystemManagerChild.h	Sat Sep 10 04:35:49 2022 +0000
@@ -12,94 +12,10 @@
 
 namespace mozilla::dom {
 
-// XXX This can be greatly simplified if bug 1786484 gets fixed.
-// See: https://phabricator.services.mozilla.com/D155351
-//      https://phabricator.services.mozilla.com/D155352
-
 class FileSystemManagerChild : public PFileSystemManagerChild {
  public:
   NS_INLINE_DECL_REFCOUNTING_WITH_DESTROY(FileSystemManagerChild, Destroy())
 
-  virtual void SendGetRootHandle(
-      mozilla::ipc::ResolveCallback<FileSystemGetHandleResponse>&& aResolve,
-      mozilla::ipc::RejectCallback&& aReject) {
-    PFileSystemManagerChild::SendGetRootHandleMsg(
-        std::forward<
-            mozilla::ipc::ResolveCallback<FileSystemGetHandleResponse>>(
-            aResolve),
-        std::forward<mozilla::ipc::RejectCallback>(aReject));
-  }
-
-  virtual void SendGetDirectoryHandle(
-      const FileSystemGetHandleRequest& aRequest,
-      mozilla::ipc::ResolveCallback<FileSystemGetHandleResponse>&& aResolve,
-      mozilla::ipc::RejectCallback&& aReject) {
-    PFileSystemManagerChild::SendGetDirectoryHandleMsg(
-        aRequest,
-        std::forward<
-            mozilla::ipc::ResolveCallback<FileSystemGetHandleResponse>>(
-            aResolve),
-        std::forward<mozilla::ipc::RejectCallback>(aReject));
-  }
-
-  virtual void SendGetFileHandle(
-      const FileSystemGetHandleRequest& aRequest,
-      mozilla::ipc::ResolveCallback<FileSystemGetHandleResponse>&& aResolve,
-      mozilla::ipc::RejectCallback&& aReject) {
-    PFileSystemManagerChild::SendGetFileHandleMsg(
-        aRequest,
-        std::forward<
-            mozilla::ipc::ResolveCallback<FileSystemGetHandleResponse>>(
-            aResolve),
-        std::forward<mozilla::ipc::RejectCallback>(aReject));
-  }
-
-  virtual void SendGetFile(
-      const FileSystemGetFileRequest& aRequest,
-      mozilla::ipc::ResolveCallback<FileSystemGetFileResponse>&& aResolve,
-      mozilla::ipc::RejectCallback&& aReject) {
-    PFileSystemManagerChild::SendGetFileMsg(
-        aRequest,
-        std::forward<mozilla::ipc::ResolveCallback<FileSystemGetFileResponse>>(
-            aResolve),
-        std::forward<mozilla::ipc::RejectCallback>(aReject));
-  }
-
-  virtual void SendResolve(
-      const FileSystemResolveRequest& aRequest,
-      mozilla::ipc::ResolveCallback<FileSystemResolveResponse>&& aResolve,
-      mozilla::ipc::RejectCallback&& aReject) {
-    PFileSystemManagerChild::SendResolveMsg(
-        aRequest,
-        std::forward<mozilla::ipc::ResolveCallback<FileSystemResolveResponse>>(
-            aResolve),
-        std::forward<mozilla::ipc::RejectCallback>(aReject));
-  }
-
-  virtual void SendGetEntries(
-      const FileSystemGetEntriesRequest& aRequest,
-      mozilla::ipc::ResolveCallback<FileSystemGetEntriesResponse>&& aResolve,
-      mozilla::ipc::RejectCallback&& aReject) {
-    PFileSystemManagerChild::SendGetEntriesMsg(
-        aRequest,
-        std::forward<
-            mozilla::ipc::ResolveCallback<FileSystemGetEntriesResponse>>(
-            aResolve),
-        std::forward<mozilla::ipc::RejectCallback>(aReject));
-  }
-
-  virtual void SendRemoveEntry(
-      const FileSystemRemoveEntryRequest& aRequest,
-      mozilla::ipc::ResolveCallback<FileSystemRemoveEntryResponse>&& aResolve,
-      mozilla::ipc::RejectCallback&& aReject) {
-    PFileSystemManagerChild::SendRemoveEntryMsg(
-        aRequest,
-        std::forward<
-            mozilla::ipc::ResolveCallback<FileSystemRemoveEntryResponse>>(
-            aResolve),
-        std::forward<mozilla::ipc::RejectCallback>(aReject));
-  }
-
   virtual void Shutdown() {
     if (CanSend()) {
       Close();
@@ -113,21 +29,6 @@
     Shutdown();
     delete this;
   }
-
- private:
-  using PFileSystemManagerChild::SendGetRootHandleMsg;
-
-  using PFileSystemManagerChild::SendGetDirectoryHandleMsg;
-
-  using PFileSystemManagerChild::SendGetFileHandleMsg;
-
-  using PFileSystemManagerChild::SendGetFileMsg;
-
-  using PFileSystemManagerChild::SendResolveMsg;
-
-  using PFileSystemManagerChild::SendGetEntriesMsg;
-
-  using PFileSystemManagerChild::SendRemoveEntryMsg;
 };
 
 }  // namespace mozilla::dom